lbrynet/daemon/Publisher.py

```diff
@@ -54,6 +55,42 @@ def create_and_publish_stream(self, name, bid, claim_dict, file_path, claim_addr
     @defer.inlineCallbacks
     def publish_stream(self, name, bid, claim_dict, claim_address=None, change_address=None):
         """Make a claim without creating a lbry file"""
+        log.info('Starting publish for %s', name)
+
+        source = claim_dict['stream']['source'].get('source', None)
+        source_type = claim_dict['stream']['source'].get('sourceType', None)
+        content_type = claim_dict['stream']['source'].get('contentType', 'application/octet-stream')
+        version = claim_dict['stream']['source'].get('version', '_0_0_1')
+
+        if source is None:
+            raise Exception("Cannot publish a stream with no source")
+
+        if source_type is None:
```

Comment:
These checks should go in lbryschema, an dictionary containing invalid values passed to `ClaimDict.load_dict` should throw a `DecodeError` or an error specific to the field (like `UnknownSourceType`)